pkg/cmd/chat.go

```diff
@@ -137,10 +133,12 @@ func (o *ChatCmdOptions) Run(cmd *cobra.Command) error {
 		}
 	}()
 	// Run TUI
+	var tuiErr error
 	if _, err := p.Run(); err != nil {
-		return fmt.Errorf("failed to run program: %w", err)
+		tuiErr = fmt.Errorf("failed to run program: %w", err)
 	}
-	return nil
+	aiAgent.Close()
```

Comment:
you may use defer for this close

Reply:
good catch, I implemented this last minute.